src/foam/nanos/notification/email/DAOEmailService.js

```diff
@@ -0,0 +1,123 @@
+/**
+ * @license
+ * Copyright 2018 The FOAM Authors. All Rights Reserved.
+ * http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+foam.CLASS({
+  package: 'foam.nanos.notification.email',
+  name: 'DAOEmailService',
+
+  documentation: 'Place generated EmailMessages into a DAO pipeline.',
+
+  implements: [
+    'foam.nanos.notification.email.EmailService'
+  ],
+
+  javaImports: [
+    'foam.dao.DAO',
+    'foam.dao.NullDAO',
+    'java.nio.charset.StandardCharsets',
+    'org.jtwig.environment.EnvironmentConfiguration',
+    'org.jtwig.environment.EnvironmentConfigurationBuilder',
+    'org.jtwig.JtwigModel',
+    'org.jtwig.JtwigTemplate',
+    'org.jtwig.resource.loader.TypedResourceLoader'
+  ],
+
+  properties: [
+    {
+      name: 'dao',
+      class: 'foam.dao.DAOProperty',
+      javaFactory: `
+        initializing.set(true);
+        String daoName = "emailMessageDAO";
+        System.out.println("DAOEmailService initializing "+daoName);
+        DAO dao = (DAO) getX().get(daoName);
+        if ( dao == null ) {
+          System.err.println("DAOEmailService DAO not found: "+daoName);
+          dao = new NullDAO();
+        }
+        initializing.set(false);
+        return dao;
+`
+    },
+  ],
+
+  axioms: [
+    {
+      name: 'javaExtras',
+      buildJavaClass: function(cls) {
+        cls.extras.push(foam.java.Code.create({
+          data:
+          `
+            protected ThreadLocal<Boolean> initializing = new ThreadLocal<Boolean>() {
```

Comment:
What is this used for?

Reply:
Now that it's all complete, it's not required. It was modelled after DAOLogMessage, but was replacing the logger which we only wanted to do once.